### DISCUSSION TITLE
Add support for macOS "Files & Folders" system services (Quick Actions)

### DISCUSSION THREAD

**Discussion by gkertai** [videovillage/community#109] [Ideas]
Most macOS applications expose their open files to [Quick Actions](https://www.macosxautomation.com/automator/services/index.html) that you can create in Automator. Apparently Screen does not do that.

I have Quick Action workflows set up for sorting & shortlisting video files, like so…
![Screenshot 2021-04-01 at 17 01 03](https://user-images.githubusercontent.com/1874498/113313794-e290dd00-930b-11eb-8acb-dc9035bc294f.png)

It would be amazing to have these available from Screen…
- for the currently open video (via the standard Services menu)
- for playlist items (via the right-click context menu)

**Comment by gregcotten** [comment]
It seems the Services menu isn't as simple as I thought... I am not confident I can use it in the context of a playlist, for example, as there is no developer-facing way to query available services for a certain context (file item, pasteboard item, etc.).

It is possible I can at least get single-file players to support specific Services. I'll look into it!

**Comment by gkertai** [comment]
Thank you for looking into it!

I've seen the [Path Finder](https://cocoatech.com/#/) app in particular to replicate *Services* menu items in its context menus, but they may be using undocumented hacks.
![Screenshot 2021-04-01 at 20 28 04](https://user-images.githubusercontent.com/1874498/113338850-f6e3d280-9329-11eb-8020-bd226354b0ce.png)


As an alternative, you might want to consider context menu items for directly setting Finder labels, tags, ratings and/or Spotlight comments. Well, in case you think it makes sense to promote the playlist as a tool for reviewing/shortlisting footage.

**Comment by gkertai** [comment]
Nevermind, I figured out a workaround. I am adding markers to my clips on the playlist, then I run a script to read the markers from the JSON and translate them to Finder tags. Easy.

Let me note that the playlist/markers combo in Screen is one of the best distinguishing features when it comes to video player apps. It makes integration into my workflows a breeze! I shall take every opportunity to promote this in professional circles.

Oh, BTW, is there a [JSON schema](https://json-schema.org/) for the `.screenplaylist` format published anywhere? My trial & error discovery has been fairly successful, but there may be hidden gems I am not yet aware of.

**Comment by gregcotten** [comment]
Glad you found a workaround! No schema as of yet, but probably no hidden gems either :)

I'm still very interested in adding some of these Finder / Spotlight features into Screen, though, so I'll keep digging.
